### ISSUE TITLE
How do I use this inside a child widget?

### ISSUE THREAD

**Issue by comeonbuddy** [diegoveloper/flutter_keyboard_actions#6]
tried to use this inside a child widget where my input is at, the button is not showing up in the right position I guess it is because my child widget doesn't have a full-screen height. 

Can you add a way to use this inside a child widget instead of the scaffold body?

**Comment by diegoveloper** [comment]
Please add a minium sample code to reproduce the issue and a screenshot if
it possible.
After that I'll review the issue.

Thanks

**Comment by diegoveloper** [comment]
Hmm could you add a minimum sample code to reproduce the issue please?

**Comment by MisterJimson** [comment]
Here are 2 examples of things I run into when using this lib.

https://github.com/MisterJimson/flutter_keyboard_actions-bugreport/tree/infiniteheight

https://github.com/MisterJimson/flutter_keyboard_actions-bugreport/tree/child-dialog

The first shows that its often easy to get infinite height issues when using this lib.

Second shows what this issue is mainly about, if you use the FormKeyboardActions widget down the Widget tree (not at root) either it doesn't display or it does but the action bar is not above the keyboard.

The reason you can't always put it near the root is that for a large app with a complex Widget tree and many different text fields in different places, its very difficult to keep the actions list up to date with the current fields on the screen.

**Comment by diegoveloper** [comment]
Yeah, I see the problem of the Dialog (child-dialog), it could be very difficult to put the keyboard bar as an Overlay because it could bother the UI /UX of the user screen.  Any idea?

**Comment by MisterJimson** [comment]
My thought would be to instruct the users of this library to always put the KeyboardActions widget at the root, and provide some way to set the Actions without a reference to the widget.

This could probably be done with a global key today, but maybe there is a better way? Not sure right now.

**Comment by stevenspiel** [comment]
Also running into this

**Comment by jayjwarrick** [comment]
One idea - how about using [Overlay](https://docs.flutter.io/flutter/widgets/Overlay-class.html) and [OverlayEntry](https://docs.flutter.io/flutter/widgets/OverlayEntry-class.html)?

`Overlay` manages an internal Stack of views, and allows floating content over all other views in the app. Every `MaterialApp` comes with one. I think this could be perfect for a keyboard bar.

Example code to float a widget just above the keyboard:

```
showOverlay() {
     entry = OverlayEntry(builder: (context) {
        return Positioned(
          bottom: MediaQuery.of(context).viewInsets.bottom, // Scaffold by default adds viewInsets.bottom to offset the keyboard and keep its body visible. So, we can use viewInsets.bottom to determine the position of the top of the keyboard.
          left: 0,
          right: 0,
          child: Container(
            color: Colors.red,
          ),
        );
      });
    OverlayState overlay = Overlay.of(context); // Get the app's overlay
    overlay.insert(entry); // Add our new floating widget. Use entry.remove() to remove later
}
```
<img width="320" alt="Screen Shot 2019-05-07 at 12 14 45 PM" src="https://user-images.githubusercontent.com/3268245/57326625-c5244b80-70c1-11e9-9545-409a112c6aed.png">



Here's a [tutorial](https://www.didierboelens.com/2018/06/how-to-create-a-toast-or-notifications-notion-of-overlay/) on Overlay I followed.

**Comment by diegoveloper** [comment]
Yep, great idea. Would you like to contribute with this package and send a PR?

**Comment by jayjwarrick** [comment]
@diegoveloper I'll look in to it!

The drawback with the overlay approach is there's no easy way to add padding to offset the thickness of the bottom bar. If I get something nice I'll send a PR 👍

**Comment by diegoveloper** [comment]
@jayjwarrick nice job! , I've already published a new version `2.0.0` , let me know if it works for you guys.  (The documentation was updated)  https://pub.dev/packages/keyboard_actions

**Comment by hyperpanthera** [comment]
> Here are 2 examples of things I run into when using this lib.
> 
> https://github.com/MisterJimson/flutter_keyboard_actions-bugreport/tree/infiniteheight
> 
> https://github.com/MisterJimson/flutter_keyboard_actions-bugreport/tree/child-dialog
> 
> The first shows that its often easy to get infinite height issues when using this lib.
> 
> Second shows what this issue is mainly about, if you use the FormKeyboardActions widget down the Widget tree (not at root) either it doesn't display or it does but the action bar is not above the keyboard.
> 
> The reason you can't always put it near the root is that for a large app with a complex Widget tree and many different text fields in different places, its very difficult to keep the actions list up to date with the current fields on the screen.

Could you tell me how you fixed the issue with infinite height?

See: #38
